[FSDP] Add unittests to test distributed checkpointing

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77368
* #77356

Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

fegin committed:
[FSDP] Add unittests to test distributed checkpointing

Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
Thanks for adding this

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Add unittests to test distributed checkpointing"


Add an unittest to test FSDP + torch.distributed._shard.checkpoint

Differential Revision: [D36175575](https://our.internmc.facebook.com/intern/diff/D36175575/)

[ghstack-poisoned]
